drivers/spi/spi_context.h

```diff
@@ -57,6 +59,20 @@ struct spi_context {
 #define SPI_CONTEXT_INIT_SYNC(_data, _ctx_name)				\
 	._ctx_name.sync = Z_SEM_INITIALIZER(_data._ctx_name.sync, 0, 1)
 
+#define SPI_CONTEXT_CS_GPIO_SPEC_ELEM(node_id, prop, idx)				\
+	{ 										\
+		.gpio_dev = DEVICE_DT_GET(DT_GPIO_CTLR_BY_IDX(node_id, prop, idx)),	\
+		.gpio_pin = DT_GPIO_PIN_BY_IDX(node_id, prop, idx),			\
+		.gpio_dt_flags = DT_GPIO_FLAGS_BY_IDX(node_id, prop, idx),		\
```

Comment:
`gpio_dt_spec` (and associated APIs) can be used instead, will make things easier

Reply:
I could have used this structure but It's not compatible with spi_cs_control structure that I'm using here.

Reply:
then, wouldn't it be better to cleanup that part, too?

Reply:
@gmarull It would. But I see that there are plenty of drivers to be modified to be `gpio_dt_spec` compatible, so I think this is a topic for another PR. Besides, I would probably first convert drivers to use `spi_dt_spec`, so after that the conversion to `gpio_dt_spec` would be probably just a change of single macro (`SPI_CS_CONTROL_PTR_DT`).

Reply:
FYI I have started converting drivers to `spi_dt_spec` in #37358 and have commits to convert the spi_cs_control to `gpio_dt_spec` here: https://github.com/csiro-wsn/zephyr/commits/210801_spi_cs

Reply:
perhaps it would be better to get spi_cs_control to gpio_dt_spec first then, and afterwards this patch

Reply:
@JordanYates When those changes will be ready? I can't see any MR with them.